rust/agents/relayer/src/msg/metadata/multisig.rs

```diff
@@ -0,0 +1,195 @@
+use async_trait::async_trait;
+use ethers::abi::Token;
+use hyperlane_core::accumulator::merkle::Proof;
+use std::collections::HashMap;
+use std::fmt::Debug;
+use std::ops::Deref;
+
+use derive_new::new;
+use eyre::Context;
+use tracing::{debug, info, instrument};
+
+use hyperlane_core::{
+    HyperlaneMessage, MultisigIsm, MultisigSignedCheckpoint, SignatureWithSigner, H256,
+};
+
+use super::base::MetadataBuilder;
+use super::BaseMetadataBuilder;
+
+#[derive(Clone, Debug, new)]
+pub struct MultisigIsmMetadataBuilder {
+    base: BaseMetadataBuilder,
+    legacy: bool,
+}
+
+impl Deref for MultisigIsmMetadataBuilder {
+    type Target = BaseMetadataBuilder;
+
+    fn deref(&self) -> &Self::Target {
+        &self.base
+    }
+}
+
+#[async_trait]
+impl MetadataBuilder for MultisigIsmMetadataBuilder {
+    #[instrument(err)]
+    async fn build(
+        &self,
+        ism_address: H256,
+        message: &HyperlaneMessage,
+    ) -> eyre::Result<Option<Vec<u8>>> {
+        const CTX: &str = "When fetching MultisigIsm metadata";
```

Comment:
Really starting to think I should make a proc macro to auto-wrap all returned results with a context.